datasync/datasync.go

```diff
@@ -1,11 +1,67 @@
 package datasync
 
 import (
+	"crypto/ecdsa"
+	"github.com/golang/protobuf/proto"
+	datasyncpeer "github.com/status-im/status-protocol-go/datasync/peer"
 	datasyncnode "github.com/vacp2p/mvds/node"
+	datasyncproto "github.com/vacp2p/mvds/protobuf"
+	datasynctransport "github.com/vacp2p/mvds/transport"
+	"go.uber.org/zap"
 )
 
 type DataSync struct {
 	*datasyncnode.Node
 	// DataSyncNodeTransport is the implemntation of the datasync transport interface
 	*DataSyncNodeTransport
+	logger         *zap.Logger
+	sendingEnabled bool
+}
+
+func New(node *datasyncnode.Node, transport *DataSyncNodeTransport, sendingEnabled bool, logger *zap.Logger) *DataSync {
+	return &DataSync{Node: node, DataSyncNodeTransport: transport, sendingEnabled: sendingEnabled, logger: logger}
+}
+
+func (d *DataSync) Add(publicKey *ecdsa.PublicKey, datasyncMessage datasyncproto.Payload) {
+	packet := datasynctransport.Packet{
+		Sender:  datasyncpeer.PublicKeyToPeerID(*publicKey),
+		Payload: datasyncMessage,
+	}
+	d.DataSyncNodeTransport.AddPacket(packet)
+}
+
+func (d *DataSync) Handle(sender *ecdsa.PublicKey, payload []byte) [][]byte {
```

Comment:
I am not sure we can treat datasync as a multiplexer from the caller perspective but if so maybe:
```suggestion
func (d *DataSync) Demux(sender *ecdsa.PublicKey, payload []byte) [][]byte {
```

Reply:
I would be a bit cautious about it, as it not only demux, there are also some side-effects involveds (the message might be replicated etc), so I feel `Demux` might be a bit misleading